saq/job.py

```diff
@@ -258,3 +268,77 @@ def replace(self, job):
         """Replace current attributes with job attributes."""
         for field in job.__dataclass_fields__:
             setattr(self, field, getattr(job, field))
+
+
+class PeriodicHeartbeat:
+    """Class to manage agent Heartbeats while Job is running"""
+
+    def __init__(self, job: Job) -> None:
+        self._running: bool = True
+        self._heartbeat_task: typing.Optional[asyncio.Task] = None
+        self.job = job
+        self.heartbeat_enabled: bool = job.heartbeat and job.heartbeat > 0
+        self.time_between_ticks = (
```

Comment:
I think "tick" has a specific meaning in other contexts. Should "tick" be "beat"?